lib/travis/build/appliances/services.rb

```diff
@@ -35,6 +31,25 @@ def apply?
           services.any?
         end
 
+        def apply_mongodb
+          sh.if "$(lsb_release -cs) != 'precise'" do
+            sh.cmd 'sudo service mongod start', assert: false, echo: true, timing: true
+          end
+          sh.else do
+            sh.cmd 'sudo service mongodb start', assert: false, echo: true, timing: true
+          end
+        end
+
+        def apply_mysql
+          sh.raw <<~BASH
+            travis_mysql_ping() {
+              mysql &>/dev/null <<<'SHOW VARIABLES like "%version%"'
+            }
+          BASH
+          sh.cmd 'sudo service mysql start', assert: false, echo: true, timing: true
+          sh.cmd 'travis_wait travis_mysql_ping', assert: false, echo: false, timing: false
```

Comment:
@meatballhat I suggest dropping the `travis_mysql_ping` function definition, and replacing this command `travis_wait travis_mysql_ping` with what I've suggested. (There should be a more succinct way to test TCP port 3306, though, than using `ruby`.)